packages/react/src/components/form/Search/SearchClear.tsx

```diff
@@ -6,27 +6,21 @@ import { setReactInputValue } from '../Combobox/utilities';
 export type SearchClearProps = Omit<ButtonProps, 'variant' | 'children'>;
 
 export const SearchClear = forwardRef<HTMLButtonElement, SearchClearProps>(
-  (props, ref) => {
-    const { 'aria-label': label = 'Tøm', onClick, ...rest } = props;
-
+  function SearchClear({ 'aria-label': label = 'Tøm', onClick, ...rest }, ref) {
     const handleClear = (
-      e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+      event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
     ) => {
-      const target = e.target;
-      let input: HTMLElement | null | undefined = null;
-
-      if (target instanceof HTMLElement)
-        input = target.closest('.ds-search')?.querySelector('input');
+      const input =
+        event.target instanceof HTMLElement &&
+        event.target.closest('.ds-search')?.querySelector('input');
 
-      if (!input) throw new Error('Input is missing');
-      /* narrow type to make TS happy */
       if (!(input instanceof HTMLInputElement))
-        throw new Error('Input is not an input element');
+        throw new Error('Input is missing');
```

Comment:
This is never gonna hit, so we could just retun from this and not throw an error.
We query select an `input`, and as the comment that was here said, it is only to narrow the type for TS